app/src/main/java/com/hover/stax/presentation/bounties/BountiesViewModel.kt

```diff
@@ -0,0 +1,103 @@
+package com.hover.stax.presentation.bounties
+
+import android.app.Application
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import com.hover.sdk.api.Hover
+import com.hover.sdk.sims.SimInfo
+import com.hover.stax.countries.CountryAdapter
+import com.hover.stax.domain.model.Bounty
+import com.hover.stax.domain.model.Resource
+import com.hover.stax.domain.use_case.bounties.GetChannelBountiesUseCase
+import com.hover.stax.domain.use_case.channels.GetPresentSimsUseCase
+import com.hover.stax.utils.Utils.getPackage
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class BountyViewModel(private val simsUseCase: GetPresentSimsUseCase, private val bountiesUseCase: GetChannelBountiesUseCase, val application: Application) : ViewModel() {
+
+    private val _countryList = MutableStateFlow<List<String>>(emptyList())
+    val countryList = _countryList.asStateFlow()
+
+    private val _sims = MutableStateFlow<List<SimInfo>>(emptyList())
+    val sims = _sims.asStateFlow()
+
+    private val _bountiesState = MutableStateFlow(BountiesState())
+    val bountiesState = _bountiesState.asStateFlow()
+
+    private val _country = MutableStateFlow(CountryAdapter.CODE_ALL_COUNTRIES)
+    val country = _country.asStateFlow()
+
+    private val onBountySelectEvent = Channel<BountySelectEvent>()
```

Comment:
Should this be an interface instead? Since it's a click or selection event?

Reply:
Not necessarily. Every interaction with on the screen can be modeled as an `event`. Thus we can make the `BountyViewModel` responsible for the state of the screen. I tried using an interface but the events weren't being propagated. It seems to only work when you provide an actual function and not an abstraction.